src/python/Services/ReqMgr/ReqMgrReader.py

```diff
@@ -6,53 +6,383 @@
+        :tries: number of tries
+        :wait: wait time between tries
+        :return: function output, deafault value o/w
+        """
+        for i in range(tries):
+            try:
+                return fcn(*fcn_pargs, **fcn_args)
+
+            except Exception as error:
+                self.logger.error(
+                    f"Failed to get to run function {fcn.__name__} with arguments {fcn_pargs} and {fcn_args} on try #{i+1} of {tries}"
+                )
+                self.logger.error(str(error))
+                time.sleep(wait)
+
+        if default:
+            return default
+        raise Exception("NoDefaultValue")
+
+    def getWorkflowSchema(
+        self,
+        wf: str,
+        drop_null: bool = False,
+        make_copy: bool = False,
+        tries: int = 2,
+        try_cache: bool = False,
+        wait: int = 0,
+    ) -> dict:
+        """
+        The function to get the schema for a given workflow
+        :param wf: workflow name
+        :param drop_null: if True, drop key with null values from the schema
+        :param make_copy: if True, return a copy of the schema
+        :param tries: number of tries for gettting a request response
+        :return: a dict
+        """
+        try:
+            return self._runWithRetries(
+                self._getWorkflowSchema,
+                [wf],
+                {"drop_null": drop_null, "make_copy": make_copy},
+                tries=tries,
+                wait=wait,
+            )
+
+        except Exception as error:
+            self.logger.error(f"Failed to get workload from reqmgr for workflow {wf}")
+            self.logger.error(str(error))
+            if try_cache:
+                self.logger.info("Will try to get workload cache")
+                return self._getWorkflowSchema(
+                    wf, drop_null=drop_null, make_copy=make_copy, from_cache=True
+                )
+
+    def _getWorkflowSchema(
+        self,
+        wf: str,
+        drop_null: bool = False,
+        make_copy: bool = False,
+        from_cache: bool = False,
+    ) -> dict:
+        """
+        The function to get the schema for a given workflow
+        :param wf: workflow name
+        :param drop_null: if True, drop key with null values from the schema
+        :param make_copy: if True, return a copy of the schema
+        :param from_cache: if True, get schema from cache
+        :return: a dict
+        """
+        result = getResponse(
+            url=self.reqmgrUrl,
+            endpoint=self.reqmgrEndpoint["couchdb" if from_cache else "req"] + wf,
+        )
+        if make_copy:
+            data = copy.deepcopy(result["result"][0][wf])
+        else:
+            data = result["result"][0][wf]
+            
+        if drop_null:
+            return {k: v for k, v in data.items() if v not in [None, "None"]}
+        return data
+
+    def getWorkflowSchemaByCampaign(
```

Comment:
Change name to `getWorkflowsByCampaign`. `Schema` is confusing in case `details` is False.